org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/DataModels/PropertyDefinition.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+/**
+ * Property Definition
+ * For more details on the TOSCA specification, visit:
+ * <a href="https://docs.oasis-open.org/tosca/TOSCA/v2.0/csd06/TOSCA-v2.0-csd06.html#93-property-definition">Property Definition</a>
+ */
+
+package org.eclipse.winery.lsp.Server.ServerCore.DataModels;
+
+import java.util.Map;
+import java.util.Optional;
+
+public record PropertyDefinition<T>(    String type,
```

Comment:
Some formatting --> spaces between opening brace and `String`

Reply:
Reading this, you will probably need two variants of the TOSCA representation:

1. A parsed one, where references are not resolved
2. A fully linked model. See https://docs.oasis-open.org/tosca/TOSCA/v2.0/csd06/TOSCA-v2.0-csd06.html#9-properties-attributes-and-parameters - here, the whole type system of TOSCA is explained. Thus, the `type` needs to be an instance of a `ToscaType` and not some `String`

I think, there will be two parsing phases: First, parse the TOSCA yaml into basic Java objects (like these you already implemented). Then, parse into full TOSCA objects having in-memory cross references to other objects.

Reply:
Using `<T>` is wrong here. This would be OK for the full TOSCA model, but not for the parsed one. 


    public record PropertyDefinition<T>(    T type,

Would be the java object for the full TOSCA model.